Added nightly build

kairveeehh opened the pull request:
solves(#90 )
created the workflow that checks for the build and release workflow on a nightly basis

kairveeehh commented:
@nirs , @jandubois  kindly review and merge .

nirs commented on .github/workflows/nightly-build.yaml:
```diff
@@ -0,0 +1,21 @@
+
+  release:
+    uses: ./.github/workflows/release.yaml  # Calls the existing release workflow
+    needs: build  # Ensures build completes first
```
We don't depend on build for testing release, so we can remove this dependency.

nirs commented on .github/workflows/nightly-build.yaml:
```diff
@@ -0,0 +1,21 @@
+    uses: ./.github/workflows/build.yaml  # Calls the existing build workflow
+
+  release:
+    uses: ./.github/workflows/release.yaml  # Calls the existing release workflow
```
This is very cool way to reuse the existing workflow!

nirs reviewed (commented):
Adding a nightly build workflow using the existing workflow is nice trick. But we really need to test all workflows periodically to catch regressions early in macOS, brew, lima, and qemu. So I think adding a schedule build to the existing workflow will be simpler.

The suggest way is better we need do something before or after running the other workflows, but we don't.

The first commit is not related, so it is better to base this pr on the current master branch.

kairveeehh commented:
> Adding a nightly build workflow using the existing workflow is nice trick. But we really need to test all workflows periodically to catch regressions early in macOS, brew, lima, and qemu. So I think adding a schedule build to the existing workflow will be simpler.
> 
> The suggest way is better we need do something before or after running the other workflows, but we don't.
> What is to be done here ? Should I add something that is to be done after the build workflow is tested? 
> The first commit is not related, so it is better to base this pr on the current master branch.
Sure will do that.

nirs commented:
> > > Adding a nightly build workflow using the existing workflow is nice trick. But we really need to test all workflows periodically to catch regressions early in macOS, brew, lima, and qemu. So I think adding a schedule build to the existing workflow will be simpler.
> > The suggest way is better we need do something before or after running the other workflows, but we don't.
> > What is to be done here ? Should I add something that is to be done after the build workflow is tested?

I think we don't need a new workflow. Instead add the schedule to the existing workflows. This will run them also periodically.

We need to be sure that the release workflow does not try to create a new release on each run. It should create a new release only when creating a tag.

nirs commented:
> ok so instead of the separate `.yaml` file i will just add the cron job to build and release files while making sure that release doesn't create a new release everytime. I hope i am getting it right?

Almost, we want to run also the lima and qemu workflows daily:
https://github.com/lima-vm/socket_vmnet/tree/master/.github/workflows

I think the release workflow is already protected by this condition:

```
if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v')
```

So running during nightly build should not be an issue.

kairveeehh commented:
> > ok so instead of the separate `.yaml` file i will just add the cron job to build and release files while making sure that release doesn't create a new release everytime. I hope i am getting it right?
> 
> Almost, we want to run also the lima and qemu workflows daily: https://github.com/lima-vm/socket_vmnet/tree/master/.github/workflows
> 
> I think the release workflow is already protected by this condition:
> 
> ```
> if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v')
> ```
> 
> So running during nightly build should not be an issue.

Ok will work in this direction , thanks.

kairveeehh commented:
> Adding a nightly build workflow using the existing workflow is nice trick. But we really need to test all workflows periodically to catch regressions early in macOS, brew, lima, and qemu. So I think adding a schedule build to the existing workflow will be simpler.
> 
> The suggest way is better we need do something before or after running the other workflows, but we don't.
> 
> The first commit is not related, so it is better to base this pr on the current master branch.

the commit is in the history but the changes have been undone , can be checked under files changed.

kairveeehh commented:
> > > The first commit is not related, so it is better to base this pr on the current master branch.
> > 
> > 
> > the commit is in the history but the changes have been undone , can be checked under files changed.
> 
> You need to rebase on top of master, dropping the first 2 commits, which should not be part of this PR now.
> 
> The best way to do this is with interactive rebase:
> 
> ```
> git rebase -i master
> ```
> 
> You will see a new editor window with this:
> 
> ```
> pick 5834e50 Updated README.md                                                                                                                                                                
> pick d9826dc added the workflow for nightly-check
> pick 54e68dd updated the workflows to run every night
> 
> # Rebase f654000..54e68dd onto f654000 (3 commands)
> #
> # Commands:
> # p, pick <commit> = use commit
> # r, reword <commit> = use commit, but edit the commit message
> # e, edit <commit> = use commit, but stop for amending
> # s, squash <commit> = use commit, but meld into previous commit
> # f, fixup [-C | -c] <commit> = like "squash" but keep only the previous
> #                    commit's log message, unless -C is used, in which case
> #                    keep only this commit's message; -c is same as -C but
> #                    opens the editor
> # x, exec <command> = run command (the rest of the line) using shell
> # b, break = stop here (continue rebase later with 'git rebase --continue')
> # d, drop <commit> = remove commit
> # l, label <label> = label current HEAD with a name
> # t, reset <label> = reset HEAD to a label
> # m, merge [-C <commit> | -c <commit>] <label> [# <oneline>]
> #         create a merge commit using the original merge commit's
> #         message (or the oneline, if no original merge commit was
> #         specified); use -c <commit> to reword the commit message
> # u, update-ref <ref> = track a placeholder for the <ref> to be updated
> #                       to this position in the new commits. The <ref> is
> #                       updated at the end of the rebase
> #
> # These lines can be re-ordered; they are executed from top to bottom.
> #
> # If you remove a line here THAT COMMIT WILL BE LOST.
> #
> # However, if you remove everything, the rebase will be aborted.
> #
> ```
> 
> Delete lines with the unwanted commits and exit saving changes.
> 
> git log will show:
> 
> ```
> commit d03c8c6e829d0671fbcf75e09921d6c941556190 (HEAD -> nightly-check)
> Author: Kairvee Vaswani <imt_2023037@iiitm.ac.in>
> Date:   Thu Feb 20 21:46:17 2025 +0530
> 
>     updated the workflows to run every night
>     
>     Signed-off-by: Kairvee Vaswani <imt_2023037@iiitm.ac.in>
> 
> commit f6540007c25c118c5d0f2172deec1f2dcb256174 (upstream/master, upstream/HEAD, master)
> Merge: faa2f74 35e34e8
> Author: Akihiro Suda <akihiro.suda.cz@hco.ntt.co.jp>
> Date:   Tue Feb 18 20:16:01 2025 +0900
> 
>     Merge pull request #117 from nirs/perf
>     
>     Improve performance test
> ```
> 
> Now force push to your repo and the pull request will be updated:
> 
> ```
> git push -f
> ```

thanks a lot for the detailed help , completed it.

nirs commented on .github/workflows/release.yaml:
```diff
@@ -3,6 +3,7 @@
 # Apache License 2.0
 
 name: Release
+
```
Unrelated change. It will be better to add such changes in separate commit. This makes it easy to revert the schedule change if needed.

nirs commented on .github/workflows/qemu.yaml:
```diff
@@ -6,6 +6,8 @@ on:
       - 'release/**'
   pull_request:
+  schedule:
+    - cron: '0 0 * * *'  
```
Any reason why we don't have the comment here? The comment is very helpful to understand this cryptic format. I'm not sure about repeating it in every file, but I think our options are:
- no comment so we don't need to repeat it
- commnet every file

nirs reviewed (commented):
Looks good, would be nice to remove the unrelated whitespace changes or separate them to another commit.

kairveeehh commented:
> > solves(#90 )
> 
> This is helpful but not enough to link to the issue. Please edit and put this at the end of the pr message:
> 
> Fixes #90
> 
> This is recognized by github and makes this pr linked to the issue. When the pr is merged, the issue will be closed.

did all the changes. 
- added comment to all the files ( had missed one by mistake) 
- removed the empty space change and amended the last commit itself.

nirs commented on .github/workflows/release.yaml:
```diff
@@ -12,8 +12,12 @@ on:
+
 permissions:
   contents: read
+
```
This is another unrelated change. I commented only on the first and hoped you will remove all the whitespace changes.

kairveeehh replied:
yup removed all of them now , really sorry for the misinterpretation from my side.

nirs commented on .github/workflows/build.yaml:
```diff
@@ -6,6 +6,9 @@ on:
   pull_request:
+  schedule:
+    - cron: '0 0 * * *'  # Runs every night at midnight UTC
+  workflow_dispatch: 
```
Why did you add workflow_dispatch and why only in this file?

nirs reviewed (approved):
Thanks!

jandubois reviewed (approved):
Thanks, LGTM
